lib/storage/metric_name.go

```diff
@@ -511,19 +471,7 @@ func MarshalMetricNameRaw(dst []byte, labels []prompbmarshal.Label) []byte {
 	dstLen := len(dst)
 	dstSize := dstLen
 	for i := range labels {
-		if i >= maxLabelsPerTimeseries {
```

Comment:
Looks like moving this check to every ingestion protocol has the following downsides:
1. Performance impact, as every ingestion protocol will have to loop through all labels to check the limits. And then loop again to unmarshal to canonical metric name
2. Maintainance impact. Now limit check needs to be called explicitly in every protocol. It is easy to forget to do so and limits won't be working.

I think we should keep checks here for reasons above. And introduce a new function that can be excpliticly called by vmagent to check the limits.